barretenberg/cpp/src/barretenberg/vm2/constraining/relations/field_gt.test.cpp

```diff
@@ -0,0 +1,263 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "barretenberg/numeric/uint256/uint256.hpp"
+#include "barretenberg/vm/avm/trace/gadgets/range_check.hpp"
+#include "barretenberg/vm2/common/aztec_types.hpp"
+#include "barretenberg/vm2/constraining/testing/check_relation.hpp"
+#include "barretenberg/vm2/generated/flavor_settings.hpp"
+#include "barretenberg/vm2/generated/relations/lookups_ff_gt.hpp"
+#include "barretenberg/vm2/simulation/events/field_gt_event.hpp"
+#include "barretenberg/vm2/simulation/field_gt.hpp"
+#include "barretenberg/vm2/simulation/lib/field_comparison.hpp"
+#include "barretenberg/vm2/simulation/testing/mock_range_check.hpp"
+#include "barretenberg/vm2/testing/fixtures.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/field_gt_trace.hpp"
+#include "barretenberg/vm2/tracegen/lib/lookup_builder.hpp"
+#include "barretenberg/vm2/tracegen/precomputed_trace.hpp"
+#include "barretenberg/vm2/tracegen/range_check_trace.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::constraining {
+namespace {
+
+using ::testing::NiceMock;
+using ::testing::TestWithParam;
+
+using tracegen::LookupIntoDynamicTableSequential;
+using tracegen::TestTraceContainer;
+
+using simulation::EventEmitter;
+using simulation::FieldGreaterThan;
+using simulation::FieldGreaterThanEvent;
+using simulation::MockRangeCheck;
+using simulation::RangeCheck;
+using simulation::RangeCheckEvent;
+
+using FF = AvmFlavorSettings::FF;
+using C = Column;
+using ff_gt = bb::avm2::ff_gt<FF>;
+
+using lookup_a_hi_range = bb::avm2::lookup_ff_gt_a_hi_range_relation<FF>;
+using lookup_a_lo_range = bb::avm2::lookup_ff_gt_a_lo_range_relation<FF>;
+
+const uint256_t TWO_POW_128 = uint256_t{ 1 } << 128;
+
+TEST(FieldGreaterThanConstrainingTest, EmptyRow)
+{
+    check_relation<ff_gt>(testing::empty_trace());
+}
+
+struct TestParams {
+    FF a;
+    FF b;
+    bool expected_result;
+};
+
+std::vector<TestParams> comparison_tests = {
+    // GT
+    TestParams{ 27, 0, true },
+    TestParams{ TWO_POW_128, 0, true },
+    TestParams{ FF::neg_one(), FF::zero(), true },
```

Comment:
FYI `FF::zero()` is being deprecated, https://github.com/AztecProtocol/aztec-packages/issues/1270. We should probably get into the habit of not using them